pkg/container-hook/bpf/execruntime.bpf.c

```diff
@@ -35,40 +35,6 @@ struct {
 	__type(value, struct record);
 } exec_args SEC(".maps");
 
-// man clone(2):
-//   If any of the threads in a thread group performs an
-//   execve(2), then all threads other than the thread group
-//   leader are terminated, and the new program is executed in
-//   the thread group leader.
-//
-// sys_enter_execve might be called from a thread and the corresponding
-// sys_exit_execve will be called from the thread group leader in case of
-// execve success, or from the same thread in case of execve failure. So we
-// need to lookup the pid from the tgid in sys_exit_execve.
-//
-// We don't know in advance which execve(2) will succeed, so we need to keep
-// track of all tgid<->pid mappings in a BPF map.
-//
-// We don't want to use bpf_for_each_map_elem() because it requires Linux 5.13.
-//
-// If several execve(2) are performed in parallel from different threads, only
-// one can succeed. The kernel will run the tracepoint syscalls/sys_exit_execve
-// for the failing execve(2) first and then for the successful one last.
-//
-// So we can insert a tgid->pid mapping in the same hash entry by adding
-// the pid in value and removing it by subtracting. By the time we need to
-// lookup the pid by the tgid, there will be only one pid left in the hash entry.
```

Comment:
I think this comment is valuable. Why not to adapt indicating why two different tracepoints need to be used for the exit?